src/components/Gallery.astro

```diff
@@ -197,6 +197,7 @@ const { images } = Astro.props;
 <script>
   const buttonsOpen = document.querySelectorAll(".gallery__button");
   const buttonsClose = document.querySelectorAll(".gallery__close");
+  const dialogs = document.querySelectorAll(".gallery__dialog");
```

Comment:
`const dialogs = document.querySelectorAll<HTMLDialogElement>(".gallery__dialog");` lets you delete all your `as HTMLDialogElement` below.